### PR TITLE
agda: bump stdlib to 2.0

### PR DESCRIPTION
**Pull request by pthariensflame**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

### CODE REVIEW

File: Formula/a/agda.rb [hunk 1]
```diff
@@ -2,14 +2,16 @@ class Agda < Formula
...
   stable do
     url "https://hackage.haskell.org/package/Agda-2.6.4.1/Agda-2.6.4.1.tar.gz"
     sha256 "23248a9b3c50c81ea4751518a66f2a6144617b6a5a9202686b54e33bc9cbd080"
 
     resource "stdlib" do
-      url "https://github.com/agda/agda-stdlib/archive/refs/tags/v1.7.3.tar.gz"
-      sha256 "91c42323fdc94d032a8c98ea9249d9d77e7ba3b51749fe85f18536dbbe603437"
+      url "https://github.com/agda/agda-stdlib.git",
+          tag:      "v2.0",
+          revision: "2b8fff10f4033b91a6df4007e4a65cb10047c89c"
```
Comment (p-linnane) [inline Formula/a/agda.rb:14]: Can we use the tarball instead of the git tag?

Comment (pthariensflame) [reply Formula/a/agda.rb:14]: We totally can, but I wasn't sure if audit liked that anymore; it's told me off for it before.

Comment (p-linnane) [reply Formula/a/agda.rb:14]: Should be fine to use the tarball then. I'll take a look if there is some kind of audit failure. Thanks!

**Comment by fxcoudert** [comment]
```
  ==> /opt/homebrew/Cellar/agda/2.6.4.1_1/bin/agda -i . -i src --html --vim README.agda
  /opt/homebrew/Cellar/agda/2.6.4.1_1/lib/agda/README.agda:1,1-1
  Cannot read file /opt/homebrew/Cellar/agda/2.6.4.1_1/lib/agda/README.agda
  Error: /opt/homebrew/Cellar/agda/2.6.4.1_1/lib/agda/README.agda: openBinaryFile: does not exist (No such file or directory)
```

**Comment by pthariensflame** [comment]
@fxcoudert Should be fixed now!

**Comment by fxcoudert** [comment]
```
  ==> /opt/homebrew/Cellar/agda/2.6.4.1_1/bin/agda -i . -i src --html --vim doc/README.agda
  /opt/homebrew/Cellar/agda/2.6.4.1_1/lib/agda/doc/README.agda:3,8-14
  The name of the top level module does not match the file name. The
  module README should be defined in one of the following files:
    /opt/homebrew/Cellar/agda/2.6.4.1_1/lib/agda/src/README.agda
    /opt/homebrew/Cellar/agda/2.6.4.1_1/lib/agda/src/README.lagda
    /opt/homebrew/Cellar/agda/2.6.4.1_1/lib/agda/README.agda
    /opt/homebrew/Cellar/agda/2.6.4.1_1/lib/agda/README.lagda
    /opt/homebrew/Cellar/agda/2.6.4.1_1/libexec/ghc-9.8.1/Agd-2.6.4.1-12156ee6/share/lib/prim/README.agda
    /opt/homebrew/Cellar/agda/2.6.4.1_1/libexec/ghc-9.8.1/Agd-2.6.4.1-12156ee6/share/lib/prim/README.lagda
```

**Comment by pthariensflame** [comment]
@fxcoudert Should be *actually* fixed this time; checked locally in several different configurations after double-checking the environment and erasing confounding config variables.

**Review by fxcoudert — APPROVED** [review]

**Pull request merged**